### PR TITLE
Use compat features to deprecate bad version statements

### PR DESCRIPTION
**Pull request by mandolaerik**
- **Use compat features to deprecate bad version statements**
- **Improve wording**
- **Use package_path from Simics base**

### CODE REVIEW

**Comment by syssimics** [comment]
Verification [#13306627](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13306627): pass

**Comment by mandolaerik** [comment]
test failure is unrelated

**Review by lwaern-intel — CHANGES_REQUESTED** [review]
This currently has parts from #279 and it shouldn't! Filter them out.

File: py/dml/compat.py [hunk 1]
```diff
@@ -99,9 +99,21 @@ class function_in_extern_struct(CompatFeature):
...
     ```
     '''
-    short = 'Disallow non-pointer function members in extern structs'
+    short = 'Allow non-pointer function members in extern structs'
     last_api_version = api_7
 
+
+@feature
+class optional_version_statement(CompatFeature):
+    '''When this compatibility feature is enabled, the `dml 1.4;`
```
Comment (lwaern-intel) [inline py/dml/compat.py:108]: > The version specification statement (e.g. `dml 1.4;`) at the start of each file is optional

File: py/dml/toplevel.py [hunk 2]
```diff
@@ -74,33 +74,41 @@ def determine_version(filestr, filename):
...
-        lineno = 1
-        column = 1
+        if compat.optional_version_statement in dml.globals.enabled_compat:
+            report(WNOVER(SimpleSite(f"{filename}:1")))
+            version = (1, 2)
+            lineno = 1
+            column = 1
+        else:
+            raise ESYNTAX(
+                SimpleSite(f"{filename}:1"), None,
```
Comment (lwaern-intel) [inline py/dml/toplevel.py:92]: Consider breaking out the identical usages of `SimpleSite` to a variable defined before the `if`.

**Review by lwaern-intel — CHANGES_REQUESTED** [review]
Approved modulo a minor quibble.

**Review by lwaern-intel — APPROVED** [review]
no really it is approved. I want you to address the quibble, but if you disagree with it, that's fine.

**Comment by syssimics** [comment]
Verification [#13346188](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13346188): pass

File: py/dml/toplevel.py (hunk 2 above)
Comment (mandolaerik) [reply py/dml/toplevel.py:92]: No need to, duplication in dying code is harmless

File: py/dml/compat.py (hunk 1 above)
Comment (mandolaerik) [reply py/dml/compat.py:108]: I would agree that this clarification would be needed if this were in a non-dying part of the spec, but this is a dying path so easy-to-read is more important than formal correctness. It's easy to understand that it refers to `dml 1.2;` in a 1.2 context.

**Comment by syssimics** [comment]
Verification [#13346831](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13346831): pass

**Pull request merged**